module/extension/Director/src/information/get_group_info.cpp

```diff
@@ -23,10 +23,39 @@ namespace module::extension {
 
     using ::extension::behaviour::GroupInfo;
 
-    GroupInfo Director::_get_group_info(const uint64_t& /*reaction_id*/, const std::type_index& type) {
+    GroupInfo Director::_get_group_info(const uint64_t& reaction_id,
+                                        const std::type_index& type,
+                                        const std::type_index& root_type) {
         std::lock_guard<std::recursive_mutex> lock(director_mutex);
+
+        uint64_t r_id = reaction_id;
+
+        // If it is a root provider, get the root provider id
+        if (!providers.contains(reaction_id)) {
+            r_id = get_root_provider(root_type)->id;
+        }
+
         if (groups.contains(type)) {
-            return GroupInfo{groups.at(type).done};
+            // Check if the task is active
+            if (groups.at(type).active_task != nullptr) {
+                // Check if the task is being run by this reaction
+                if (groups.at(type).active_task->requester_id == r_id) {
+                    return GroupInfo{GroupInfo::RunState::RUNNING, groups.at(type).done};
+                }
+            }
+
+            // Check if this task is in the reaction's task list
+            // If it is, then it is waiting to be run
+            if (providers.contains(r_id)) {
```

Comment:
You know this is true from your earlier checks, you can probably make a `auto&` for it right after the root_type stuff